tests/integration/test_events.py

```diff
@@ -571,59 +572,122 @@ def received(q_urls):
         self.sfn_client.delete_state_machine(stateMachineArn=state_machine_arn)
 
     def test_api_destinations(self):
+
+        token = short_uid()
+        bearer = "Bearer %s" % token
+
         class HttpEndpointListener(ProxyListener):
             def forward_request(self, method, path, data, headers):
                 event = json.loads(to_str(data))
                 events.append(event)
-                return 200
+                paths_list.append(path)
+                auth = headers.get("Api") or headers.get("Authorization")
+                if auth not in headers_list:
+                    headers_list.append(auth)
+
+                return requests_response(
+                    {
+                        "access_token": token,
+                        "token_type": "Bearer",
+                        "expires_in": 86400,
+                    }
+                )
 
         events = []
+        paths_list = []
+        headers_list = []
+
         local_port = get_free_tcp_port()
         proxy = start_proxy(local_port, update_listener=HttpEndpointListener())
         wait_for_port_open(local_port)
-
         events_client = aws_stack.create_external_boto_client("events")
-        connection_arn = events_client.create_connection(
-            Name="TestConnection",
-            AuthorizationType="BASIC",
-            AuthParameters={"BasicAuthParameters": {"Username": "user", "Password": "pw"}},
-        )["ConnectionArn"]
-
-        # create api destination
-        dest_name = "d-%s" % short_uid()
         url = "http://localhost:%s" % local_port
-        result = self.events_client.create_api_destination(
-            Name=dest_name,
-            ConnectionArn=connection_arn,
-            InvocationEndpoint=url,
-            HttpMethod="POST",
-        )
 
-        # create rule and target
-        rule_name = "r-%s" % short_uid()
-        target_id = "target-{}".format(short_uid())
-        pattern = json.dumps({"source": ["source-123"], "detail-type": ["type-123"]})
-        self.events_client.put_rule(Name=rule_name, EventPattern=pattern)
-        self.events_client.put_targets(
-            Rule=rule_name,
-            Targets=[{"Id": target_id, "Arn": result["ApiDestinationArn"]}],
-        )
+        auth_types = [
```

Comment:
side note: If and when we migrate this test to `pytest`, we can look into parameterizing this test using the `@pytest.mark.parametrize` decorator.. 👍  (no change required for this PR, could be done in a future iteration)